src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1525,1898 @@ interface MPSMatrixMultiplication {
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnConvolutionNode), Name = "MPSCNNBinaryConvolutionNode")]
+	interface MPSCnnBinaryConvolutionNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:scaleValue:type:flags:")]
+		MPSCnnBinaryConvolutionNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+
+		[Export ("initWithSource:weights:scaleValue:type:flags:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnBinaryConvolutionNode), Name = "MPSCNNBinaryFullyConnectedNode")]
+	interface MPSCnnBinaryFullyConnectedNode
+	{
+		[Static]
+		[Export ("nodeWithSource:weights:scaleValue:type:flags:")]
+		MPSCnnBinaryFullyConnectedNode FromSource (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+
+		[Export ("initWithSource:weights:scaleValue:type:flags:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, IMPSCnnConvolutionDataSource weights, float scaleValue,MPSCnnBinaryConvolutionType type,MPSCnnBinaryConvolutionFlags flags);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnConvolutionNode), Name = "MPSCNNConvolutionTransposeNode")]
+	interface MPSCnnConvolutionTransposeNode
+	{
+		[Static]
+		[Export ("nodeWithSource:convolutionState:weights:")]
+		MPSCnnConvolutionTransposeNode FromSource (MPSNnImageNode sourceNode, [NullAllowed]MPSCnnConvolutionStateNode convolutionState, IMPSCnnConvolutionDataSource weights);
+
+		[Export ("initWithSource:convolutionState:weights:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, [NullAllowed]MPSCnnConvolutionStateNode convolutionState, IMPSCnnConvolutionDataSource weights);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSCNNNeuronNode")]
+	[DisableDefaultCtor]
+	interface MPSCnnNeuronNode
+	{
+		[Export ("a")]
+		float A { get; }
+
+		[Export ("b")]
+		float B { get; }
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnNeuronNode), Name = "MPSCNNNeuronAbsoluteNode")]
+	interface MPSCnnNeuronAbsoluteNode
+	{
+		[Static]
+		[Export ("nodeWithSource:")]
+		MPSCnnNeuronAbsoluteNode FromSource (MPSNnImageNode sourceNode);
+
+		[Export ("initWithSource:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnNeuronNode), Name = "MPSCNNNeuronELUNode")]
+	interface MPSCnnNeuronELUNode
+	{
+		[Static]
+		[Export ("nodeWithSource:a:")]
+		MPSCnnNeuronELUNode FromSource (MPSNnImageNode sourceNode, float a);
```

Comment:
Methods should have verbs: `CreateFromSource` (and all the other ones below as well).